pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaSinkWrappingProducer.java

```diff
@@ -0,0 +1,188 @@
+    private final static List<PartitionInfo> partitionInfos = ImmutableList.of(info);
+
+    public KafkaSinkWrappingProducer(SinkConnector connector,
+                                     SinkTask task,
+                                     Schema defaultKeySchema,
+                                     Schema defaultValueSchema,
+                                     SinkContextSim sinkContext,
+                                     SinkTaskContextSim taskContext) {
+        this.connector = connector;
+        this.task = task;
+        this.defaultKeySchema = defaultKeySchema;
+        this.defaultValueSchema = defaultValueSchema;
+        this.sinkContext = sinkContext;
+        this.taskContext = taskContext;
+    }
+
+    public static Set<TopicPartition> getPartitions() {
+        return partitions;
+    }
+
+    @Override
+    public void initTransactions() {
+        throw new UnsupportedOperationException("not supported");
+    }
+
+    @Override
+    public void beginTransaction() throws ProducerFencedException {
+        throw new UnsupportedOperationException("not supported");
+    }
+
+    @Override
+    public void sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> map, String s)
+            throws ProducerFencedException {
+        throw new UnsupportedOperationException("not supported");
+    }
+
+    @Override
+    public void commitTransaction() throws ProducerFencedException {
+        throw new UnsupportedOperationException("not supported");
+    }
+
+    @Override
+    public void abortTransaction() throws ProducerFencedException {
+        throw new UnsupportedOperationException("not supported");
+    }
+
+    private SinkRecord toSinkRecord(ProducerRecord<K, V> producerRecord) {
+        int partition = producerRecord.partition() == null ? 0 : producerRecord.partition();
+        Schema keySchema = defaultKeySchema;
+        Schema valueSchema = defaultValueSchema;
+
+        if (producerRecord instanceof ProducerRecordWithSchema) {
+            ProducerRecordWithSchema rec = (ProducerRecordWithSchema) producerRecord;
+            keySchema = rec.getKeySchema();
+            valueSchema = rec.getValueSchema();
+        }
+
+        SinkRecord sinkRecord = new SinkRecord(producerRecord.topic(),
+                partition,
+                keySchema,
+                producerRecord.key(),
+                valueSchema,
+                producerRecord.value(),
+                taskContext.getAndIncrementOffset(),
+                producerRecord.timestamp(),
+                TimestampType.NO_TIMESTAMP_TYPE);
+        return sinkRecord;
+    }
+
+    @Override
+    public Future<RecordMetadata> send(ProducerRecord<K, V> producerRecord) {
+        sinkContext.throwIfNeeded();
+        task.put(Lists.newArrayList(toSinkRecord(producerRecord)));
+        return CompletableFuture.completedFuture(null);
```

Comment:
is it possible to return a CompletableFuture that completes when the SinkRecord has been successfully processed ?

otherwise Pulsar thinks that the record has been correctly "sent" (processed) but this is not true.

the same comment applies to the other methods below

Reply:
@eolivelli I'll add flush() immediately after task.put() to guarantee that for now.
The proper way would be implementation of batching support (and policies based on time/number of pending records) but I'd prefer to postpone this as a "performance improvement" until we agree on functional completeness and/or hear some feedback that this is needed.